test/gtest/common/mem_buffer.cc

```diff
@@ -92,23 +92,25 @@ const std::vector<ucs_memory_type_t>&  mem_buffer::supported_mem_types()
 
 void mem_buffer::set_device_context()
 {
-    static __thread bool device_not_set = true;
+    static __thread bool device_set = false;
+
+    if (device_set) {
+        return;
+    }
 
-    if (device_not_set) {
-        try {
 #if HAVE_CUDA
-            CUDA_CALL(cudaSetDevice(0), "set device context");
+    if (is_cuda_supported()) {
+        cudaSetDevice(0);
+    }
 #endif
+
 #if HAVE_ROCM
```

Comment:
no need for `HAVE_RCOM`, it is already checked in `is_rocm_supported()`

Reply:
it's wrapping hipSetDevice

Reply:
sorry, I see
thanks